client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -226,7 +231,20 @@ public class ControllerImpl implements Controller {
     private final Supplier<Long> requestIdGenerator = RandomFactory.create()::nextLong;
 
     private final long timeoutMillis;
-    
+
+    /**
+     * Maximum time that segment endpoint details are held in the cache.
+     */
+    private final Duration segmentEntryExpirationTime = Duration.ofMinutes(5);
+    /**
+     * Maximum number of segment endpoint entries in the cache.
+     */
+    private final int maxCacheSize = 1000;
```

Comment:
Is this number decided on any specific reason?

Reply:
Yes, since it is a cache, so it was advised to start with this size. @tkaitchuck , please correct if any changes are required to this.

Reply:
This is a reasonable number. We can change it if a larger or smaller number proves desireable.